src/module_ncio.f90

```diff
@@ -793,11 +802,13 @@ function create_dataset(filename, dsetin, copy_vardata, paropen, nocompress, mpi
        endif
     enddo
   end function create_dataset
-
+  !> close a netcdf file, deallocate members of dataset object.
+  !! if optional error return code errcode is not specified,
+  !! program will stop if a nonzero error code returned by the netcdf lib.
+  !! @param filename: filename for netCDF Dataset.
```

Comment:
Always insert a blank line of documentation before the first param tag, like this:

```
!> Some description in a complete sentence.
!!
!! @param a description of a
!! @author Freddy Fortran

```
Note the line with just "!!" - that's what I mean by a blank line of documentation before the first param.